### PR TITLE
enable etag on latest version by resolving latest

### PR DESCRIPTION
**Pull request by YannanGao-gs**
enable etag on latest version by resolving latest

Screenshot 2024-09-23 at 11 13 46 AM

### CODE REVIEW

File: legend-depot-entities-api/src/main/java/org/finos/legend/depot/services/api/entities/EntitiesService.java [hunk 1]
```diff
@@ -52,4 +52,6 @@ default List<ProjectVersionEntities> getDependenciesEntitiesByClassifier(String
     {
         return getDependenciesEntitiesByClassifier(Arrays.asList(new ProjectVersion(groupId, artifactId, versionId)), classifier, transitive, includeOrigin);
     }
+
+    String resolveVersionId(String groupId, String artifactId, String versionId);
```
Comment (gs-gunjan) [inline legend-depot-entities-api/src/main/java/org/finos/legend/depot/services/api/entities/EntitiesService.java:56]: does not seem like the correct place, plus we already have a function like so, can we think of something better

**Comment by gs-gunjan** [comment]
looks good to me

File: legend-depot-entities-services/src/test/java/org/finos/legend/depot/server/resources/TestEntitiesResource.java [hunk 2]
```diff
@@ -135,7 +137,7 @@ public void canGetMetrics() throws InterruptedException
         metricsHandler.persist(metricsRegistry);
 
         Assert.assertNotNull(metricsStore.get("examples.metadata", "test", "2.3.0").get(0).getLastQueryTime());
-        Assert.assertEquals(2, metricsStore.getAllStoredEntities().size());
+        Assert.assertEquals(3, metricsStore.getAllStoredEntities().size());
```
Comment (MauricioUyaguari) [inline legend-depot-entities-services/src/test/java/org/finos/legend/depot/server/resources/TestEntitiesResource.java:140]: why did the test change here ?

**Comment by MauricioUyaguari** [comment]
why didnt we also fix the other apis that use latest such as the dependnecy entities api  ?

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**